600.java

```diff
@@ -65,29 +75,30 @@ public static void main(String[] args){
 	for(int i = 0; i < n; i++){
 		nums[i] = Integer.parseInt(strs[i]);
 	}
-	int[] order = new int[n];
+	int[] order = new int[n + 1];
 	for(int i = 0; i < n; i++){
```

Comment:
The only way I can think of is to use TreeMap to keep the insertion order and map number with its order, can't really do without additional array or LinkedList to cache number and its insertion order.